blocks/library/quote/index.js

```diff
@@ -20,6 +20,36 @@ registerBlock( 'core/quote', {
 		citation: html( 'footer' )
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'block',
+				blocks: [ 'core/text' ],
+				transform: ( { content } ) => {
+					return {
+						value: content,
+					};
+				},
+			},
+		],
+		to: [
+			{
+				type: 'block',
+				blocks: [ 'core/text' ],
+				transform: ( { value, citation } ) => {
+					if ( citation ) {
+						return new Error(
+							'Quote citation would be lost on transform.'
```

Comment:
I don't love this because switching a quote block to a text block and back would leave the citation as a separate paragraph.